spring-integration-stomp/src/main/java/org/springframework/integration/stomp/outbound/StompMessageHandler.java

```diff
@@ -104,9 +106,11 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(final Message<?> message) throws Exception {
-		if (!this.isRunning()) {
+		if (!this.isRunning() || !this.stompSessionManager.isConnected()) {
+			this.running = false;
+			this.stompSession = null;
 			throw new MessageDeliveryException(message, "The StompMessageHandler [" + getComponentName() +
-					"] hasn't been connected to StompSession. Check the state of [" + this.stompSessionManager + "]");
+					"] isn't connected to StompSession. Check the state of [" + this.stompSessionManager + "]");
 		}
```

Comment:
Well, the difference here with the other protocols that STOMP session is a bit *statefull* object and we can't use different sessions for the `inbound` and `outbound` adapters like we can do that with the AMQP or MQTT.

The STOMP Session works like XMPP Connection, meaning that we start a conversation with the server. Yes, we can use it as an one-way (only `inbound` or `outbound`) communication channel, but that doesn't mean that e should have different methods to connect from the adapters.

Although, I think, I can find the way for the `StompMessageHandler` to "connect on request" rather than reject the Message. That may look like "wait for connection", but not `MessageDeliveryException`.

The `StompInboundChannelAdapter` will still work as is, similar to the `MqttPahoMessageDrivenChannelAdapter` or `SimpleMessageListenerContainer` .